dev-util/hip/hip-6.1.0.ebuild

```
@@ -0,0 +1,157 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DOCS_BUILDER="doxygen"
+DOCS_DEPEND="media-gfx/graphviz"
+ROCM_SKIP_GLOBALS=1
+
+inherit cmake docs flag-o-matic llvm rocm
+
+LLVM_MAX_SLOT=17
+
+TEST_PV=${PV}
+
+DESCRIPTION="C++ Heterogeneous-Compute Interface for Portability"
+HOMEPAGE="https://github.com/ROCm/clr"
+SRC_URI="https://github.com/ROCm/clr/archive/refs/tags/rocm-${PV}.tar.gz -> rocm-clr-${PV}.tar.gz
+	https://github.com/ROCm/HIP/archive/refs/tags/rocm-${PV}.tar.gz -> hip-${PV}.tar.gz
+	test? ( https://github.com/ROCm/hip-tests/archive/refs/tags/rocm-${TEST_PV}.tar.gz -> hip-test-${TEST_PV}.tar.gz )"
+S="${WORKDIR}/clr-rocm-${PV}/"
+TEST_S="${WORKDIR}/hip-tests-rocm-${TEST_PV}/catch"
+
+KEYWORDS="~amd64"
+LICENSE="MIT"
+SLOT="0/$(ver_cut 1-2)"
+
+RESTRICT="!test? ( test )"
+IUSE="debug test"
+
+DEPEND="
+	>=dev-util/rocminfo-5
+	sys-devel/clang:${LLVM_MAX_SLOT}
+	dev-libs/rocm-comgr:${SLOT}
+	>=dev-libs/rocr-runtime-5.6
+	x11-base/xorg-proto
+	virtual/opengl
+"
+BDEPEND="test? ( dev-util/hipcc )"
+RDEPEND="${DEPEND}
+	dev-util/hipcc
+	dev-perl/URI-Encode
+	sys-devel/clang-runtime:=
+	>=dev-libs/roct-thunk-interface-5"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-5.7.1-disable-stack-protector.patch"
+	"${FILESDIR}/${PN}-5.7.1-no_asan_doc.patch"
+	"${FILESDIR}/${PN}-6.1.0-extend-isa-compatibility-check.patch"
+)
+
+hip_test_wrapper() {
+	local CMAKE_USE_DIR="${TEST_S}"
+	local BUILD_DIR="${TEST_S}_build"
+	cd "${TEST_S}" || die
+	$@
+}
+
+src_prepare() {
+	# hipamd is itself built by cmake, and should never provide a
+	# FindHIP.cmake module.
+	rm -r "${WORKDIR}"/HIP-rocm-${PV}/cmake/FindHIP* || die
+
+	cmake_src_prepare
+
+	local PATCHES=(
+		"${FILESDIR}"/hip-test-6.0.2-hipcc-system-install.patch
+		"${FILESDIR}"/hip-test-5.7.1-remove-incompatible-flag.patch
+		"${FILESDIR}"/hip-test-6.1.0-disable-hipKerArgOptimization.patch
+	)
+	hip_test_wrapper cmake_src_prepare
```

```suggestion
	use test  && hip_test_wrapper cmake_src_prepare
```
otherwise fails with default `USE="-test"`

---

I'm not putting `use test &&` here because conditional patching is discouraged[1], so the patches to build tests won't break silently when bumping versions with `-test`

https://devmanual.gentoo.org/ebuild-writing/misc-files/patches/index.html#conditional-patching